cpp/client/ssl_client.cc

```diff
@@ -96,8 +96,6 @@ SSLClient::SSLClient(const string& server, uint16_t port, const string& ca_dir,
 
 SSLClient::~SSLClient() {
   Disconnect();
-  if (ctx_ != NULL)
-    SSL_CTX_free(ctx_);
```

Comment:
Would you want to make `verify_args_.verifier` a `unique_ptr`, while you're at it?